### DISCUSSION TITLE
Is it possible to change the trigger key for different features?

### DISCUSSION THREAD

**Discussion by matthewfurr** [johnlindquist/kit#773] [Q&A] [answered]
Is it possible to change that keys trigger different features. For example, I would like `.` to bring up the App Launcher instead of the current trigger `;`.

Current triggers are:
```
; = App Launcher
/ = File Browser
> = Terminal
< = Clipboard History
1-9 = Calculator
, = Scratch Pad
. = File Search
' = Snippets
[ = Templates
" = Word API
- = System Commands
= = Dev Tools
: = Emoji Picker
~ = Google Suggest
\ = Google
```

Thanks in advance!

**Comment by johnlindquist** [comment, marked as answer]
Yup!

If you look in `~/.kit/handler/` you'll see a list of named scripts that "handle" each character.

You can override these handlers by creating scripts with the same names in your main kenv. For example, to change the `.` behavior, you would create a `period-handler`:

https://user-images.githubusercontent.com/36073/170851768-ac268e0e-85bd-469e-b4cc-dd4ea229f928.mp4

So you'd want to copy the contents of `~/.kit/main/app-launcher.js` to `~/.kenv/scripts/period-handler.js`. You can install this script and run it to do it for you 😇

[Open convert-app-launcher-to-period-handler in Script Kit](https://scriptkit.com/api/new?name=convert-app-launcher-to-period-handler&url=https://gist.githubusercontent.com/johnlindquist/0d2f074a4d324412d32ccc87a848d35b/raw/164e5912f9c16ec87e3f7dc0ba1c00ea6d34dffe/convert-app-launcher-to-period-handler.ts")

```js
// Name: Convert App Launcher to Period Handler

import "@johnlindquist/kit"

await copyFile(
  home(".kit", "main", "app-launcher.js"), 
  home(".kenv", "scripts", "period-handler.js")
)
```

Let me know if you need any help!

## Note

On the first run on typing `.`, hit `cmd+enter` to regenerate the icons in the proper dir.

**Reply by matthewfurr** [reply]
Thanks @johnlindquist! I found the handler directory after creating this post, but decided it was best to not mess around in the `~/.kit/` directory. I'll give this a go and report back if I have any issues, but this looks fairly straight forward. Thanks again!

**Reply by johnlindquist** [reply]
This doesn't change anything in the `~/.kit` dir, it's solely adding a script in your scripts dir that's named `period-handler.js`. Copy/pasting the contents of the App Launcher is just making in the script that you want it to be.

Also, you may want to use the metadata to exclude it from your main menu search:
```js
// Exclude: true
```

**Reply by hakubo** [reply]
Is it possible to add a direct shortcut to one of these?
like `cmd shift v` to open clipboard history?

**Reply by johnlindquist** [reply]
@hakubo This oughta do it:

[Open testing-shortcut-clipboard-history in Script Kit](https://scriptkit.com/api/new?name=testing-shortcut-clipboard-history&url=https://gist.githubusercontent.com/johnlindquist/11904df03a2f0dfed143da826a5435e6/raw/8fafb97fabca5ef13a9ea8cd6a21be29d86140f1/testing-shortcut-clipboard-history.ts")

```js
// Name: Testing Shortcut Clipboard History
// Shortcut: cmd shift v

import "@johnlindquist/kit"

await run(kitPath("main", "clipboard-history.js"))
```

Also, we're starting up a discord if you'd like to come hang :)
https://discord.gg/pY84ERdK

**Reply by hakubo** [reply]
`await main("clipboard-history");` - that seems to work!

**Reply by johnlindquist** [reply]
Just fyi, `main` is kind of an "internal global" used by the Kit SDK, so it's not exposed by the TypeScript types. TypeScript would complain that `main` is missing, but if you're not using TypeScript, then no worries :D
